jupyterlite_sphinx/jupyterlite_sphinx.py

```diff
@@ -990,15 +989,38 @@ def jupyterlite_build(app: Sphinx, error):
             jupyterlite_contents = [jupyterlite_contents]
 
         # Expand globs in the contents strings
-        jupyterlite_contents = [
-            match
-            for pattern in jupyterlite_contents
-            for match in glob.glob(pattern, recursive=True)
-        ]
-
         contents = []
-        for content in jupyterlite_contents:
-            contents.extend(["--contents", content])
+        for pattern in jupyterlite_contents:
+            pattern_path = Path(pattern)
+
+            if pattern_path.is_absolute():
+                base_path = pattern_path.parent
+                glob_pattern = pattern_path.name
+            else:
+                # For relative paths, we'll resolve them relative
+                # to app.srcdir
+                base_path = Path(app.srcdir) / pattern_path.parent
+                glob_pattern = pattern_path.name
+
+            if any(c in str(glob_pattern) for c in "*?[]"):
+                matched_paths = list(base_path.glob(glob_pattern))
+            else:
+                full_path = base_path / glob_pattern
+                matched_paths = [full_path] if full_path.exists() else []
```

Comment:
```suggestion
            matched_paths = base_path.glob(glob_pattern)
```

I'm not sure its worth adding all of these extra lines just to avoid calling `glob` when there are no special characters in the pattern. I can't imagine it makes any appreciable difference in runtime. Also, you don't need to convert it into a list because you're just going to iterate through it later anyway.